### PR TITLE
FAKE PR, DO NOT MERGE: Config API review 4.1 -> 4.2

### PR DESCRIPTION
**Pull request by sttts**
This is a fake PR enabling review of the 4.2 changes and their OpenAPI annotations.

/hold

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -118,6 +138,67 @@ type AWSPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// AzurePlatformStatus holds the current status of the Azure infrastructure provider.
+type AzurePlatformStatus struct {
+	// resourceGroupName is the Resource Group for new Azure resources created for the cluster.
+	ResourceGroupName string `json:"resourceGroupName"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @abhinavdahiya is this required?

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -118,6 +138,67 @@ type AWSPlatformStatus struct {
...
+}
+
+// GCPPlatformStatus holds the current status of the Google Cloud Platform infrastructure provider.
+type GCPPlatformStatus struct {
+	// resourceGroupName is the Project ID for new GCP resources created for the cluster.
+	ProjectID string `json:"projectID"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @abhinavdahiya is this required?

File: config/v1/types_infrastructure.go [hunk 3]
```diff
@@ -118,6 +138,67 @@ type AWSPlatformStatus struct {
...
+
+	// region holds the region for new GCP resources created for the cluster.
+	Region string `json:"region"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @abhinavdahiya is this required?

File: config/v1/types_infrastructure.go [hunk 4]
```diff
@@ -118,6 +138,67 @@ type AWSPlatformStatus struct {
...
+}
+
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// apiServerInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @bcrochet are these three new fields optional?

File: config/v1/types_infrastructure.go [hunk 5]
```diff
@@ -118,6 +138,67 @@ type AWSPlatformStatus struct {
...
+	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+}
+
+// OpenStackPlatformStatus holds the current status of the OpenStack infrastructure provider.
+type OpenStackPlatformStatus struct {
+	// apiServerInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @tomassedovic are these 4 all optional?

File: config/v1/types_scheduling.go [hunk 6]
```diff
@@ -48,6 +49,16 @@ type SchedulerSpec struct {
...
+	// MastersSchedulable allows masters nodes to be schedulable. When this flag is
+	// turned on, all the master nodes in the cluster will be made schedulable,
+	// so that workload pods can run on them. The default value for this field is false,
+	// meaning none of the master nodes are schedulable.
+	// Important Note: Once the workload pods start running on the master nodes,
+	// extreme care must be taken to ensure that cluster-critical control plane components
+	// are not impacted.
+	// Please turn on this field after doing due diligence.
+	// +optional
+	MastersSchedulable bool `json:"mastersSchedulable"`
```
Comment (sttts) [inline config/v1/types_scheduling.go]: @ravisantoshgudimetla @damemi should this be omitempty?

File: config/v1/types_infrastructure.go (hunk 4 above)
Comment (bcrochet) [reply config/v1/types_infrastructure.go]: Not if using the BareMetal platform. BareMetalPlatformStatus is marked as optional. If that needs to propagate downwards, then I guess yes.

**Pull request closed without merging**